internal/binder/function/funcs_str.go

```diff
@@ -357,4 +358,30 @@ func registerStrFunc() {
 		val:   ValidateOneStrArg,
 		check: returnNilIfHasAnyNil,
 	}
+	builtins["hex2dec"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			arg0 := strings.TrimPrefix(cast.ToStringAlways(args[0]), "0x")
```

Comment:
Do not use ToStringAlways, it will convert all types. We should only accept string type here.